File: frontend/lib/types/document.ts
export type DocumentStatus = "processing" | "ready" | "failed";

export interface DocumentSummary {
  id: string;
  filename: string;
  file_type: string;
  file_size_bytes: number;
  page_count: number | null;
  status: DocumentStatus;
  error_message: string | null;
  created_at: string;
}

export interface DocumentPage {
  page_number: number;
  total_pages: number;
  text: string;
}

File: backend/app/schemas/document.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.document import DocumentStatus


class DocumentBase(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    filename: str
    file_type: str
    file_size_bytes: int
    status: DocumentStatus
    error_message: str | None
    created_at: datetime
    updated_at: datetime


class DocumentListItem(DocumentBase):
    pass


class DocumentDetail(DocumentBase):
    pass


class DocumentCreateResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    filename: str
    status: DocumentStatus
    created_at: datetime
